Feature: Add URI fetching related triples and serialization in different formats

imadbourouche opened the pull request.

syphax-bouazzouni commented on Gemfile:
```diff
@@ -39,3 +40,5 @@ end
 gem 'goo', github: 'ontoportal-lirmm/goo', branch: 'development'
 gem 'sparql-client', github: 'ontoportal-lirmm/sparql-client', branch: 'master'
+
+gem "dip", "~> 7.7"
```
no need to add dip, just do locally `gem install dip`

syphax-bouazzouni commented on lib/ontologies_linked_data/models/base.rb:
```diff
@@ -10,7 +10,7 @@ class Base < Goo::Base::Resource
       def save(*args)
         write_permission_check(*args)
-        super(*args)
+        super(*args) rescue binding.pry
```
remove this

syphax-bouazzouni commented on test/solr/configsets/term_search/conf/schema.xml:
```diff
@@ -441,9 +441,9 @@
     <!-- A text field that only splits on whitespace for exact matching of words -->
```
reset the changes in this file

syphax-bouazzouni commented on test/test_case.rb:
```diff
@@ -38,7 +38,7 @@
 end
 
-require_relative '../config/config'
+require_relative '../config/config.test'
```
do change this line, just do in local `cp config/config.test.rb config/config.rb`

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+      
+        class Resource < LinkedData::Models::Base
+           
+            def fetch_related_triples(graph, id)
```
put this as private

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+                return hashes
+            end
+
+            def stringify_hash(hashes)
```
put this as private

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+                model_schema
+            end
+
+            def get_type(value)
```
put this as private

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+            end 
+            
+            # http://example.org/:TestPredicate11 => :test_predicate_11, http://example.org/
+            def namespace_predicate(property_url)
```
put this as private

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+    module Models
+      
+        class Resource < LinkedData::Models::Base
+           
```
add a constructor 

```ruby 
def initialize(id, graph)
   @id = id 
  @graph = graph 
  @hash = fetch_related_triples(graph, id)
end

```

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+            def to_turtle(hashes, options = {})
+                LinkedData::Serializers.serialize(hashes, LinkedData::MediaTypes::TURTLE)
+            end
+
```
add 
```ruby

def to_hash
    @hash 
end
```

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+                LinkedData::Serializers.serialize(hashes, LinkedData::MediaTypes::NTRIPLES)
+            end
+
+            def to_turtle(hashes, options = {})
```
remove arguments, use instance variable `@hash`

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+                LinkedData::Serializers.serialize(hashes, LinkedData::MediaTypes::XML)
+            end
+            
+            def to_ntriples(hashes, options = {})
```
remove arguments, use the instance variable `@hash`

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+                LinkedData::Serializers.serialize(hashes, LinkedData::MediaTypes::JSON)
+            end
+
+            def to_xml(hashes)
```
remove arguments, use instance variable `@hash`

syphax-bouazzouni commented on lib/ontologies_linked_data/models/resource.rb:
```diff
@@ -0,0 +1,96 @@
+                return [match[:namespace],  match[:id]] if match
+            end 
+
+            def to_json(hashes)
```
remove arguments, use instance variable `@hash`

imadbourouche committed:
Add raptor library to parse ntriples data

imadbourouche committed:
Add resource model to fetch id related triples and serialize it

imadbourouche committed:
Add and inhance xml, ntriples, turtle and json serializers

imadbourouche committed:
Updating rdf version in goo project

imadbourouche committed:
updating resource model

imadbourouche committed:
Adding tests for resource model and serializers

syphax-bouazzouni committed:
update the resource test to have a more complete data to test (array, bnodes, typed values)

syphax-bouazzouni committed:
re-implement xml serializer using RDF/XML parser instead of Raptor

syphax-bouazzouni committed:
implement array handelling of resource to_object

imadbourouche committed:
Enhance and refactor serializers ntriples, turtle and xml

imadbourouche committed:
Enhance and refactor serializers ntriples, turtle and xml

imadbourouche committed:
Handle blank nodes and reverse triples

- handle blank nodes
- fetch reverse triples
- generate random name for models in to_object, because when two model created the same time one overrides the other
- call the new serializer JSONLD and RDF_XML

imadbourouche committed:
Impliment new serializers jsonld and rdf_xml

- impliment jsonld serializer that uses json-ld library
- revert changes in xml.rb file to the original implimentation, and put the new implimentation in rdf_xml.rb file
- Add the media types :jsonld and :rdf_xml

imadbourouche committed:
Add json-ld gem

imadbourouche committed:
Enhance the test resource

- Add some cases to the data tests
- refactor the test of the serializers formats

imadbourouche committed:
Fix test for fetch-related triples and json

syphax-bouazzouni committed:
Merge branch 'development' into feature/sparql-fetch-related-triples

syphax-bouazzouni committed:
Merge branch 'feature/sparql-fetch-related-triples' of github.com:ontoportal-lirmm/ontologies_linked_data into feature/sparql-fetch-related-triples

syphax-bouazzouni committed:
clean and refactor the resource serializer code

* Removed unused methods
* Extracted duplicated code in methods
* Removed skip from the tests
